mlrun/datastore/targets.py

```diff
@@ -93,6 +93,46 @@ def get_default_prefix_for_source(kind):
     return get_default_prefix_for_target(kind)
 
 
+def validate_target_paths_for_engine(targets, engine):
+    """Validating that target paths are suitable for the required engine.
+    validate for single file targets only (parquet and csv).
+
+    spark:
+        cannot be a single file path (e.g - ends with .csv or .pq)
+
+    storey:
+        if csv - must be a single file.
+        if parquet - in case of partitioned it must be a directory,
+                     else can be both single file or directory
+    """
+    for base_target in targets:
+        if (
+            base_target.kind == TargetTypes.parquet
+            or base_target.kind == TargetTypes.csv
+        ):
+            target = get_target_driver(base_target)
+            is_single_file = (
+                hasattr(target, "is_single_file") and target.is_single_file()
```

Comment:
No need to check `hasattr` here, since both CSV and Parquet are known to have this method. Besides, I'm not sure the logic is right here - if the class doesn't have an `is_single_file` API, it's assumed to be a directory (`is_single_file = False`)?